### ISSUE TITLE
BatchNorm on axis=-1 is very slower than axis=1

### ISSUE THREAD

**Issue by chinakook** [apache/incubator-mxnet#10095]
As profiler shows, the "nn.BatchNorm(axis=-1)" will cause 10x lower speed on my application than "nn.BatchNorm(axis=1)".
However, We often need a "nn.BatchNorm(axis=-1)" after "nn.Dense(flatten=False)".

**Comment by chinakook** [comment]
Yes, it's 4D and sometimes 3D.
I wonder that why it's different, "NHWC" and "NCHW" will be treated diffrently?

**Comment by chinakook** [comment]
I got it. 
As I've tested, Tensorflow's batchnorm on the last axis is OK. It may not use cudnn.

**Comment by eric-haibin-lin** [comment]
Yeah TF doesn’t use CUDNN. We’re waiting for CUDNN 7.2 which has lots of improvements on NHWC
